### PR TITLE
add azure-cloud-provider charm page

### PR DESCRIPTION
**Pull request by addyess**

### CODE REVIEW

File: pages/k8s/azure-cloud-provider.md [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+If you install **Charmed Kubernetes** [using the Juju bundle][install],
+you can add the azure-cloud-provider at the same time by using the following
+overlay file ([download it here][asset-azure-cloud-overlay]):
+
+```yaml
+description: Charmed Kubernetes overlay to add native Azure support.
+applications:
+  kubernetes-control-plane:
+    options:
+      register-with-taints: ""
```
Comment (kwmonroe) [inline pages/k8s/azure-cloud-provider.md:40]:
```suggestion
```

small tweak to bring in-line with [the overlay](https://github.com/charmed-kubernetes/bundle/pull/854/files)

File: pages/k8s/azure-cloud-provider.md [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+  - kubernetes-control-plane:external-cloud-provider
+- - azure-cloud-provider:azure-integration
+  - azure-integrator:clients
+```
+
+To use this overlay with the **Charmed Kubernetes** bundle, it is specified
+during deploy like this:
+
+```bash
+juju deploy charmed-kubernetes --overlay azure-cloud-overlay.yaml --trust
```
Comment (kwmonroe) [inline pages/k8s/azure-cloud-provider.md:64]: just curious, is `--trust` needed here if the overlay specifies it?

File: pages/k8s/azure-cloud-provider.md [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+
+You can then verify this works by loading the described IP address (on port
+8080!) in a browser.
+
+For more configuration options and details of the permissions which the integrator uses,
+please see the [charm page][azure-cloud-provider].
+
+## Azure load-balancers for the control plane
+
+With revision 1015 and later of the `kubernetes-control-plane` charm, Charmed
```
Comment (kwmonroe) [inline pages/k8s/azure-cloud-provider.md:175]: Does `revision 1015` mean anything useful after we moved to tracks in charmhub?  Possibly better to say `With 1.25/stable and later`?

**Review by kwmonroe — COMMENTED** [review]
Abstaining here as I'm not 100% on my comments.  Take 'em for what they're worth (about $3.50).

File: pages/k8s/azure-cloud-provider.md (hunk 2 above)
Comment (addyess) [reply pages/k8s/azure-cloud-provider.md:63]: i'm at like 85% it's required.  I don't think a bundle will deploy with a trusted charm unless the deploy command is run with trust

**Comment by evilnick** [comment]
why is this a new page and not an edit of the existing page?

**Comment by addyess** [comment]
> why is this a new page and not an edit of the existing page?

So, this is a new charm separate from `azure-integrator`.  Is that what you're thinking?

**Comment by evilnick** [comment]
@addyes - yes - there is already an azure-integration  page currently linked up in the navuigation etc. I know the charm is new but it replaces the old one, right? So, just overwrite the current page rather than adding a new one

**Comment by addyess** [comment]
> it replaces the old one, right

actually, it amends the existing integration one.  It doesn't yet replace it.  This charm is intended to be a subordinate of the azure-integrator charm

**Comment by evilnick** [comment]
> > it replaces the old one, right
> 
> actually, it amends the existing integration one. It doesn't yet replace it.

But however that works, we don't need more than one page for it so we. The pages aren't per charm ( not anymore anyhow), so the current azure-integration page should be the one people follow. If the individual charm needs docs they should be on ch

**Review by evilnick — APPROVED** [review]
Thanks for all the work on this :cake:

**Pull request merged**